api/endpoints/internal.py

```
@@ -25,10 +25,22 @@
 import logging
 from services.parsing_service import extract_text_from_pdf, extract_text_from_docx
 from services.chunking_service import chunk_text
-from services.embedding_service import embed_texts
+from services.embedding_service import embed_texts, embed_texts_batched
 # VectorDBClient intentionally not imported by default here; integrate in production
 
 
+class BatchEmbeddingRequest(BaseModel):
+    texts: List[str] = Field(..., description="List of texts to embed")
```

Ruff (0.14.1)

Undefined name `List`

(F821)